### PR TITLE
[CPU] Mask rcnn does not work via HETERO

### PR DESCRIPTION
**Pull request by egorshul**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *60378*

### CODE REVIEW

**Review by apankratovantonp — CHANGES_REQUESTED** [review]
What if operations are supported only after `Transformation()`?
The problem should be found in `ngraph` transformations!

**Comment by egorshul** [comment]
@apankratovantonp The HETERO plugin receives a list of operations without transformations. The CPU plugin after your changes gets a list of operations after transformations. One of the Unsqueeze nodes disappears after the transformations and the plugin will not execute it, so it is not in the list of supported nodes. But in fact, the CPU supports this operation, so we save the current node to the list of supported ones. If you have any questions, let's discuss offline.

**Comment by apankratovantonp** [comment]
Yes, the CPU plugin should check operations after transformations! And then restore the information about nodes that produce transformed operations using `ngraph::getFusedNamesVector`. If some ngraph pass lost this info we have to find this pass and fix it. 
`Unsqueeze` node does not disappear. It is *fused* into some other node. So the name of the `Unsqueeze` should be placed into that other node runtime info.

**Comment by egorshul** [comment]
@apankratovantonp please review the PR.

**Review by apankratovantonp — APPROVED** [review]

File: inference-engine/src/transformations/src/transformations/common_optimizations/nop_elimination.cpp [hunk 1]
```diff
@@ -160,6 +159,7 @@ static bool replace_squeeze_unsqueeze(const std::shared_ptr<Node>& node) {
 
     // skip if reshape is nop
     if (reshape->get_input_partial_shape(0).same_scheme(shape_ps)) {
+        copy_runtime_info({input, node->output(0).get_node_shared_ptr()}, node->output(0).get_node_shared_ptr());
```
Comment (dmitry-gorokhov) [inline inference-engine/src/transformations/src/transformations/common_optimizations/nop_elimination.cpp:162]: Shouldn't we copy runtime info on else branch as well?

Comment (egorshul) [reply inference-engine/src/transformations/src/transformations/common_optimizations/nop_elimination.cpp:162]: The replace_output_update_name and replace_node_update_name methods copy runtime info. An additional copy_runtime_info is required in an exceptional situation.

**Review by dmitry-gorokhov — APPROVED** [review]

**Comment by dmitry-gorokhov** [comment]
@egorshul Please rebase the PR on latest master

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**